src/scripts/condor_submit.sh

```diff
@@ -201,19 +201,27 @@ submit_file_environment="#"
 
 if [ "x$environment" != "x" ] ; then
 # Input format is suitable for bourne shell style assignment. Convert to
-# old condor format (no double quotes in submit file).
-# FIXME: probably it's better to convert everything into the 'new' Condor
-# environment format.
+# new condor format to avoid errors  when things like LS_COLORS (which 
+# has semicolons in it) get captured
     eval "env_array=($environment)"
     submit_file_environment=""
     for  env_var in "${env_array[@]}"; do
         if [ "x$submit_file_environment" == "x" ] ; then
-            submit_file_environment="environment = "
+            submit_file_environment="environment = \" ${env_var}"
         else
-            submit_file_environment="$submit_file_environment;"
+            # check for spaces in env_var
+            pattern=" "
+            if [[ $env_var =~ $pattern ]]; then
+              fixed_env_var="${env_var}'"
+              fixed_env_var=`echo ${fixed_env_var} | sed -e "s|=|='|"`
+              echo "12 -- ${fixed_env_var}" >> ~/foo
+              submit_file_environment="${submit_file_environment} ${fixed_env_var} "
+            else
+              submit_file_environment="${submit_file_environment} ${env_var} "
```

Comment:
Does this handle the semicolons issue mentioned in the comment about `LS_COLORS`?

Seems like the escaping happens only when there are spaces.